install/deploy/istio-deploy.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Add istio official repo
+add_repo(){
+  REPO="https://storage.googleapis.com/istio-release/releases/1.3.3/charts/"
+  helm repo add istio $REPO
+
+  STATUS_CMD=`echo $?`
+  CHECK_REPO_CMD=`helm repo list | grep $REPO | wc -l`
+  echo "$STATUS_CMD"
+  echo "$CHECK_REPO_CMD"
+  while [[ $STATUS_CMD != 0 && $CHECK_REPO_CMD -ge 1 ]]
+  do
+    sleep 5
+    helm repo add istio $REPO
+
+    STATUS_CMD=`echo $?`
+    CHECK_REPO_CMD=`helm repo list | grep $REPO | wc -l`
+  done
+}
+
+# Create istio-system namespace
+create_namespace() {
+  kubectl create ns istio-system
+
+  STATUS_CMD=`echo $?`
+  while [[ $STATUS_CMD != 0 ]]
+  do
+    sleep 5
+    kubectl create ns istio-system
+    STATUS_CMD=`echo $?`
+  done
+}
+
+# Create CRD need for istio
+create_crd() {
+  helm install istio-init istio/istio-init -n istio-sytem
```

Comment:
I prefer to use `helm template` instead of `helm install`. Because if there's something wrong with YAMLs generated by Helm, we can check it out with template files.
You can generate YAML by issuing `helm template`, then apply them.

Reply:
My idea is to use the `install` , I think we don't have to customize istio as a CI process, and the official istio chart, there won't be any serious mistakes in theory.

Reply:
Trust me. You will face a log of wired problems when to deploy in CI. If you want to know what's wrong about them, YAML is a crucial point to figure it out. 
And don't trust istio. It will cheat on you sometime, believe me, I suffer from it many times. especially upgrade it.